storage/backend/ceph/internals.go

```diff
@@ -1,26 +1,32 @@
 package ceph
 
 import (
+	"encoding/json"
 	"fmt"
 	"os/exec"
 	"regexp"
 	"strings"
 
-	"encoding/json"
-
 	"github.com/contiv/volplugin/storage"
 
 	log "github.com/Sirupsen/logrus"
 )
 
 func (c *Driver) mapImage(do storage.DriverOptions) (string, error) {
-	var rbdMaps map[string] struct {
-		Pool string `json:"pool"`
-		Name string `json:"name"`
+	var rbdMaps map[string]struct {
+		Pool   string `json:"pool"`
+		Name   string `json:"name"`
 		Device string `json:"device"`
 	}
 
-	blkdev, err := exec.Command("rbd", "map", do.Volume.Name, "--pool", do.Volume.Params["pool"]).Output()
+	poolName := do.Volume.Params["pool"]
+
+	if err := exec.Command("rbd", "lock", "add", do.Volume.Name, do.Volume.Name, "--pool", poolName).Run(); err != nil {
```

Comment:
It's been a few weeks since I wrote this, but I am pretty sure that the unmount driver API call will cause the unlock to happen on the same host; but I will verify before submitting updates for the rest.